pkg/kubelet/cm/dra/manager_test.go

```diff
@@ -0,0 +1,405 @@
+		},
+		Status: resourcev1alpha2.ResourceClaimStatus{
+			DriverName: driverName,
+			Allocation: &resourcev1alpha2.AllocationResult{
+				ResourceHandles: []resourcev1alpha2.ResourceHandle{
+					{Data: "test-data", DriverName: driverName},
+				},
+			},
+			ReservedFor: []resourcev1alpha2.ResourceClaimConsumerReference{
+				{UID: "test-reserved"},
+			},
+		},
+	}
+
+	if _, err := fakeKubeClient.ResourceV1alpha2().ResourceClaims(pod.Namespace).Create(context.Background(), resourceClaim, metav1.CreateOptions{}); err != nil {
+		t.Fatalf("failed to create ResourceClaim: %+v", err)
+	}
+
+	manager.cache.add(&ClaimInfo{ClaimInfoState: state.ClaimInfoState{ClaimName: "test-resource-claim", Namespace: "test-namespace"}})
+
+	// Call the UnprepareResources method
+	if err := manager.UnprepareResources(pod); err != nil {
+		t.Fatalf("failed to unprepare resources: %v", err)
+	}
+
+	// Check that the cache has been updated correctly
+	claimName := resourceclaim.Name(pod, &pod.Spec.ResourceClaims[0])
+	claimInfo := manager.cache.get(claimName, pod.Namespace)
+	if claimInfo != nil {
+		t.Fatalf("claimInfo still found in cache after calling UnprepareResources")
+	}
+
+	// Allow the fake DRA Driver gRPC server to shutdown gracefully
+	close(stopCh)
+}
+
+func TestGetContainerClaimInfos(t *testing.T) {
+	cache, err := newClaimInfoCache(t.TempDir(), draManagerStateFileName)
+	if err != nil {
+		t.Fatalf("error occur:%v", err)
+	}
+	manager := &ManagerImpl{
+		cache: cache,
+	}
+
+	resourceClaimName := "test-resource-claim-1"
+	resourceClaimName2 := "test-resource-claim-2"
+	// Create a test pod and container with resource claims
+	pod := &v1.Pod{
+		Spec: v1.PodSpec{
+			ResourceClaims: []v1.PodResourceClaim{
+				{
+					Name:   "claim1",
+					Source: v1.ClaimSource{ResourceClaimName: &resourceClaimName},
+				},
+				{
+					Name:   "claim2",
+					Source: v1.ClaimSource{ResourceClaimName: &resourceClaimName2},
+				},
+			},
+		},
+	}
+	container := &v1.Container{
+		Resources: v1.ResourceRequirements{
+			Claims: []v1.ResourceClaim{
+				{
+					Name: "claim1",
+				},
+				{
+					Name: "claim2",
+				},
+			},
+		},
+	}
+
+	// Add test fakeClaimInfos to the cache
+	manager.cache.add(&ClaimInfo{ClaimInfoState: state.ClaimInfoState{ClaimName: "test-resource-claim-1"}})
+	manager.cache.add(&ClaimInfo{ClaimInfoState: state.ClaimInfoState{ClaimName: "test-resource-claim-2"}})
+
+	// Call the GetContainerClaimInfos() function
+	fakeClaimInfos, err := manager.GetContainerClaimInfos(pod, container)
+
+	// Check the results
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(fakeClaimInfos))
+	assert.Equal(t, "test-resource-claim-1", fakeClaimInfos[0].ClaimInfoState.ClaimName)
+	assert.Equal(t, "test-resource-claim-2", fakeClaimInfos[1].ClaimInfoState.ClaimName)
+
+	// Test with a missing claimInfo
+	manager.cache.delete("claim2", "default")
+	_, err = manager.GetContainerClaimInfos(pod, container)
+	assert.NoError(t, err)
+}
```

Comment:
I can see a lot of similarity in these tests. Can we combine them into the set of test cases and run with t.Run, like it's done in other tests, e.g. [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/cpumanager/cpu_assignment_test.go#L28:L126).

Reply:
ok let me try to rework that part

Reply:
done in 04cfefa673c